android/app/src/main/java/com/coboltforge/dontmind/multivnc/PointerInputHandler.java

```diff
@@ -343,10 +343,11 @@ public boolean onTouchEvent(MotionEvent e) {
 
             // modify MotionEvent to support Samsung S Pen Event and activate rightButton accordingly
             // if Samsung S Pen is not present or reports false, check for right mouse button
-            boolean rightButton = spenActionConvert(e) || e.isButtonPressed(MotionEvent.BUTTON_SECONDARY);
+            boolean isSPen = spenActionConvert(e);
+            boolean isRightButton = e.isButtonPressed(MotionEvent.BUTTON_SECONDARY);
 
-            vncCanvasActivity.vncCanvas.processPointerEvent(e, true, rightButton);
-            vncCanvasActivity.vncCanvas.panToMouse();
```

Comment:
`panToMouse()` is gone. why is there a need to touch this here logic?

Reply:
I removed `panToMouse()` because [sendPointerEvent()](https://github.com/bk138/multivnc/blob/bc5f64eeb4a81632bd8dda76b5e2f68b82395dcc/android/app/src/main/java/com/coboltforge/dontmind/multivnc/VNCConn.java#L509) already calls is for every event. It is redundant here so I thought it may be a leftover from some prior refactoring.


This logic need to be changed so that it will only trigger for API < 23 (and SPen actions). For API >= 23, clicks are handled in `onGenericMotionEvent()`. And even though we are handling `ACTION_BUTTON_PRESS` & `ACTION_BUTTON_RELEASE` events, Android still sends `ACTION_UP` & `ACTION_DOWN`  events here. We need to ignore them if they are coming from a mouse.

Reply:
Alright, understood.